### PR TITLE
CMCL-1514: Use new analytics API for 2023.2

### PR DESCRIPTION
**Pull request by glabute**
### Purpose of this PR

Old analytics API is deprecated in 2023.2, and generates console warnings.
Use the new API.

### Testing status

Don't know how to test if the new analytics calls are actually doing what we think they are.

- [ ] Added an automated test
- [ ] Passed all automated tests
- [ ] Manually tested

### CODE REVIEW

File: com.unity.cinemachine/Editor/Utility/CinemachineEditorAnalytics.cs [hunk 1]
```diff
@@ -11,13 +14,38 @@ static class CinemachineEditorAnalytics
...
+        const string k_VcamsOnPlayEventName = "cm_vcams_on_play";
+
         // register an event handler when the class is initialized
         static CinemachineEditorAnalytics()
         {
             EditorApplication.playModeStateChanged += SendAnalyticsOnPlayEnter;
         }
 
+#if UNITY_2023_2_OR_NEWER
+        [AnalyticInfo(eventName: k_CreateVcamEventName, vendorKey: k_VendorKey,
```
Comment (AntoineCharton) [inline com.unity.cinemachine/Editor/Utility/CinemachineEditorAnalytics.cs:27]: I don't see it when creating a VCam from add component is it expected?

Comment (glabute) [reply com.unity.cinemachine/Editor/Utility/CinemachineEditorAnalytics.cs:27]: It never worked from AddComponent.  It's only invoked from the entries in the Cinemachine menu.  Does it work correctly when using the menu?

Comment (AntoineCharton) [reply com.unity.cinemachine/Editor/Utility/CinemachineEditorAnalytics.cs:27]: Yes!

**Review by AntoineCharton — APPROVED** [review]

**Comment by glabute** [comment]
@AntoineCharton Did you check EnterPlaymode with multiple vcams?  It should record an event that lists all the vcams in the scene, with stuff about each.  Those are the events I'm worried about (changed implementation from List to Array)

**Comment by AntoineCharton** [comment]
@glabute 
Let me know if you have other ideas of things to check.
Sep-18-2023 16-29-43
Sep-18-2023 16-29-32

**Pull request merged**